cdap-app-fabric/src/test/java/co/cask/cdap/internal/app/runtime/spark/SparkProgramRunnerTest.java

```diff
@@ -306,6 +297,61 @@ public void apply() throws Exception {
   }
 
   @Test
+  public void testSparkWithTimePartitionedFileSetAndDSArgs() throws Exception {
+    testSparkWithTimePartitionedFileSetAndDSArgs(SparkAppUsingFileSet.class, SparkAppUsingFileSet.JavaCharCount.class);
```

Comment:
I had it that way earlier but then changed it as this one looked cleaner to me since it separates outt the test cases. But yes you are right the trade off longer run time is way too high.